selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/intelligent/CommandsEditor.java

```diff
@@ -324,85 +324,152 @@ public void waitTabIsNotPresent(String nameOfFile) {
   }
 
+  /**
+   * Waits until specified {@code expectedText} is displayed in the "Goal" field.
+   *
+   * @param expectedText text which should be displayed
+   */
   public void waitTextIntoGoalField(String expectedText) {
     seleniumWebDriverHelper.waitText(nameGoalField, expectedText, REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
   }
 
+  /**
+   * Opens "Goal" drop-down list and clicks on specified {@code goalName}.
+   *
+   * @param goalName item's visible name in the "Goal" drop-down
+   */
   public void selectGoalNameIntoCommandEditor(String goalName) {
     seleniumWebDriverHelper.waitAndClick(iconDropDownGoal, REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
     seleniumWebDriverHelper.waitVisibility(goalDropDown, REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
     seleniumWebDriverHelper.waitAndClick(By.xpath(format(NAME_IN_GOAL_DROP_DOWN, goalName)));
   }
 
+  /**
+   * Clicks on specified {commandsEditorType} in the "Commands Editor".
+   *
+   * @param commandsEditorType command editor's type which defined in {@link CommandsEditorType}
+   */
   public void setFocusIntoTypeCommandsEditor(String commandsEditorType) {
     seleniumWebDriverHelper.waitAndClick(
         By.xpath(commandsEditorType), REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
     editor.waitActive();
   }
 
+  /**
+   * Waits until specified {@code expectedText} is displayed in the popup with macro description.
+   *
+   * <p>Note! Popup with description opens after single click on item in the macro autocomplete
+   * list.
+   *
+   * @param expectedText text which should be displayed
+   */
   public void waitTextIntoDescriptionMacrosForm(String expectedText) {
     seleniumWebDriverHelper.waitVisibility(
         By.xpath(format(DESCRIPTION_MACROS, expectedText)), REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
   }
 
-  public void selectMacrosLinkInCommandsEditor(String macrosLinkType) {
-    seleniumWebDriverHelper.waitAndClick(By.xpath(macrosLinkType), REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
+  /**
+   * Waits visibility and clicks on "Macros" link with specified {@code macroLinkType} in the
+   * commands editor.
+   *
+   * @param macroLinkType type of the "Macros" link, defined in {@link CommandsMacrosLinkType}
+   */
+  public void selectMacroLinkInCommandsEditor(String macroLinkType) {
+    seleniumWebDriverHelper.waitAndClick(By.xpath(macroLinkType), REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
     waitCommandsMacrosIsOpen();
   }
 
+  /** Waits until "Command Macros" form is opened. */
   public void waitCommandsMacrosIsOpen() {
     seleniumWebDriverHelper.waitVisibility(
         By.xpath(COMMAND_MACROS_FORM), REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
   }
 
+  /** Waits until "Command Macros" form is closed. */
   public void waitCommandsMacrosIsClosed() {
     seleniumWebDriverHelper.waitInvisibility(
         By.xpath(COMMAND_MACROS_FORM), REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
   }
 
+  /**
+   * Types and waits specified {@code text} to the "Search" field in the "Command Macros" form.
```

Comment:
This JavaDoc will look simpler without additional note, IMHO. Note can be emphasized in first sentence, for example:
`Sets specified {@code text} to the "Search" field in the "Command Macros" form and waits until it appears.`